spyderlib/utils/introspection/fallback_plugin.py

```diff
@@ -44,17 +46,20 @@ def get_completions(self, info):
             items += module_completion(info.line, [info.filename])
             return [(i, '') for i in sorted(items)]
         elif info.obj:
-            base = info.obj
-            tokens = set(info.split_words(-1))
-            items = [item for item in tokens if
-                     item.startswith(base) and len(item) > len(base)]
-            if '.' in base:
-                start = base.rfind('.') + 1
-            else:
-                start = 0
-
-            items = [i[start:len(base)] + i[len(base):].split('.')[0]
-                     for i in items]

```

Comment:
I think we should leave `word` completions because it's useful for prose files like Markdown and Rest.